transformer/external/executabletransformer.go

```diff
@@ -136,33 +144,49 @@ func (t *Executable) Transform(newArtifacts []transformertypes.Artifact, already
 				SrcPath:  "",
 				DestPath: common.DefaultSourceDir,
 			})
-		} else {
-			path := ""
-			if a.Paths != nil && a.Paths[artifacts.ServiceDirPathType] != nil {
-				path = a.Paths[artifacts.ServiceDirPathType][0]
-			}
-			stdout, stderr, exitcode, err := t.Env.Exec(append(t.ExecConfig.TransformCMD, path))
-			if err != nil {
-				if errors.Is(err, &environment.EnvironmentNotActiveError{}) {
-					logrus.Debugf("%s", err)
-					continue
-				}
-				logrus.Errorf("Transform failed %s : %s : %d : %s", stdout, stderr, exitcode, err)
-				continue
-			} else if exitcode != 0 {
-				logrus.Debugf("Transform did not succeed %s : %s : %d : %s", stdout, stderr, exitcode, err)
-				continue
-			}
-			logrus.Debugf("%s Transform succeeded in %s : %s, %s, %d", t.Config.Name, t.Env.Decode(path), stdout, stderr, exitcode)
-			stdout = strings.TrimSpace(stdout)
-			var output transformertypes.TransformOutput
-			err = json.Unmarshal([]byte(stdout), &output)
-			if err != nil {
-				logrus.Errorf("Error in unmarshalling json %s: %s.", stdout, err)
+		}
+	} else {
+		transformerInputDirPath := filepath.Join(t.Env.TempPath, transformerInputDir)
```

Comment:
Since the temp path is one per transformer/environment, we might want to create another temp subdirectory below the temppath and put the transformer input below it.

Reply:
done.